rpc/ethereum/namespaces/eth/api.go

```diff
@@ -780,13 +798,14 @@ func (e *PublicAPI) GetTransactionReceipt(hash common.Hash) (map[string]interfac
 		return nil, fmt.Errorf("failed to decode tx: %w", err)
 	}
 
-	msg, err := evmtypes.UnwrapEthereumMsg(&tx)
-	if err != nil {
-		e.logger.Debug("invalid tx", "error", err.Error())
-		return nil, err
+	msg := tx.GetMsgs()[msgIndex]
```

Comment:
check out of bounds?

Reply:
I think it should be fine since the `msgIndex` is inferred from tx events, added a comment about it.